pkg/controller/controller.go

```diff
@@ -207,11 +198,26 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	capacity := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	volSizeBytes := capacity.Value()
 
+	controllerCreateSecrets := map[string]string{}
+	secret := v1.SecretReference{}
+	// get secrets if StorageClass specifies it
+	if sn, ok := options.Parameters[secretNameKey]; ok {
+		ns := "default"
+		if len(options.Parameters[secretNamespaceKey]) != 0 {
+			ns = options.Parameters[secretNamespaceKey]
+		}
+		controllerCreateSecrets = getCredentialsFromSecret(p.client, sn, ns)
```

Comment:
`getCredentialsFromSecret` should be tested for error.  If `storageClass` has secrets, then the code tries to access the secret and cannot, it should fail as an error and not silently return empty secret credentials.

Reply:
@vladimirvivien we had that discussion with @saad-ali and agreed (please correct me if I am wrong) that failing on absent secret, especially since it is not mandatory, should not be done at the provisioner level, it should be left to the plugin to take a final decision.